src/Database/Connection.php

```diff
@@ -269,20 +317,25 @@ public function getDriver(): DriverInterface
      */
     public function connect(): bool
     {
-        try {
-            return $this->_driver->connect();
-        } catch (MissingConnectionException $e) {
-            throw $e;
-        } catch (Throwable $e) {
-            throw new MissingConnectionException(
-                [
-                    'driver' => App::shortName(get_class($this->_driver), 'Database/Driver'),
-                    'reason' => $e->getMessage(),
-                ],
-                null,
-                $e
-            );
+        $connected = true;
+        foreach ([self::ROLE_READ, self::ROLE_WRITE] as $role) {
```

Comment:
Don't really like connecting both drivers when potentially only 1 could be needed. `ReconnectStrategy` is the only place where `Connection::connect()` is used. So in 5.x we should explore changes to avoid connecting both the drivers.

Reply:
Agreed, but I wasn't sure how complicated that would be so didn't change in this PR.